meow/user_profile/views.py

```diff
@@ -12,22 +12,127 @@
 import json
 # Create your views here.
 
+@api_login_required()
+def themeStar(request):
+    user = request.user
+    if request.method == "POST":
+        req_data = json.loads(request.body)
+        theme_id = req_data['id']
+        themes = Theme.objects.all()
+        star_theme = themes.get(pk=theme_id)
+        user_associated = User.objects.get(username=user.username)
+        user_associated.starred_themes.add(star_theme)
+        starred_themes_id = user_associated.starred_themes.values_list('pk', flat=True)
+        print("Starred Themes ID")
+        starred_themes_id = list(starred_themes_id)
+        return JsonResponse(starred_themes_id, safe=False, status=200)
+    elif request.method == "PUT":
+        req_data = json.loads(request.body)
+        theme_id = req_data['id']
+        themes = Theme.objects.all()
+        #theme we want to remove the relation
+        unstar_theme = themes.get(pk=theme_id)
+        user_associated = User.objects.get(username=user.username)
+        user_associated.starred_themes.remove(unstar_theme)
+        starred_themes_id = user_associated.starred_themes.values_list('pk', flat=True)
+        starred_themes_id = list(starred_themes_id)
+        return JsonResponse(starred_themes_id, safe=False, status=200)
+
+
+@api_login_required()
+def themeAdd(request):
```

Comment:
ahhh so for these functions... this is the prefered way of doing it 

https://www.django-rest-framework.org/tutorial/3-class-based-views/

Class based views handle a lot of the details for you so it makes the code more readable and easier to write.
additionally, you can use the serializers to get an object from the request.body! So for this one it would be something like `theme_serializer = ThemeSerializer(data=request.data)`. Then you can simply use `theme_serializer.is_valid()` to check if the theme is valid. And to save it, you just do `theme_serializer.save()`